src/screens/SurveyDetailScreen.tsx

```diff
@@ -149,12 +149,14 @@ export const SurveyDetailScreen = ({ route }: Props) => {
               />
             )}
           </WrapperWithOrientation>
-          <CommentSection
-            archived={archived}
-            comments={survey.comments}
-            scrollViewRef={scrollViewRef}
-            surveyId={surveyId}
-          />
+          {!!survey.canComment && (
```

Comment:
could we avoid `!!` as `canComment` is typed as boolean?

Reply:
maybe better to keep them for some `null` or `undefined` cases?

Reply:
i typed the `Survey` type to have it as an optional value, because i did not know if we will get `null` from the server under any circumstances. (e.g. it does not get set when it is off at creation, or the db migration for backward compatability was not completed in time/successfully)

Reply:
yes, was my second thought as well even when i think this cannot happen. let's be more safe here.